### PR TITLE
Update help texts to accurately reflect current AzCopy functionality

### PR DESCRIPTION
**Pull request by pranavmalik-msft**
### Description
This Pull Request aims to enhance the usability and accessibility of AzCopy by refining the help text provided in the current release. The goal is to offer clearer instructions on how to effectively utilize and configure AzCopy, thereby minimizing confusion among users. This effort aligns with our commitment to continuously improve the user experience and ensure seamless interactions with the AzCopy tool.

### CODE REVIEW

File: cmd/helpMessages.go [hunk 1]
```diff
@@ -313,6 +313,21 @@ Set the environment variable AZCOPY_SPA_CERT_PASSWORD to the certificate's passw
...
    --certificate-path is mandatory when doing cert-based service principal auth.
 
+Log in using a Device:
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE=DEVICE to initiate Device login.
+   - azcopy login
+   Please note that you will be provided with a code to authenticate via a web browser. For example, you may see a message like this:
+   To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code AWRKVXE8P.
+
+Log in using Managed Identity
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE to MSI.
```
Comment (vibhansa-msft) [inline cmd/helpMessages.go:323]: Can we tab them and align under the managed iden header.

File: cmd/helpMessages.go [hunk 2]
```diff
@@ -525,21 +548,29 @@ selected file count and size:
...
 
+Run a benchmark test that uploads 1000 files, each 100 KiB in size, and creates folders to divide up the data:
+
+   - azcopy bench "https://[account].blob.core.windows.net/[container]?<SAS>" --file-count 1000 --size-per-file 100K --number-of-folders 5
+ 
 Run a benchmark test that downloads existing files from a target
 
    - azcopy bench --mode='Download' "https://[account].blob.core.windows.net/[container]?<SAS?"
 
+Run a download benchmark with the default parameters and cap the transfer rate at 500 Mbps:
```
Comment (vibhansa-msft) [inline cmd/helpMessages.go:559]: why are we adding these samples ? what it a ask from any customer or something?

Comment (pranavmalik-msft) [reply cmd/helpMessages.go:559]: these are just some more help-texts for a better coverage of features

File: cmd/helpMessages.go [hunk 3]
```diff
@@ -323,6 +323,21 @@ Set the environment variable AZCOPY_SPA_CERT_PASSWORD to the certificate's passw
    Please treat /path/to/my/cert as a path to a PEM or PKCS12 file-- AzCopy does not reach into the system cert store to obtain your certificate.
    --certificate-path is mandatory when doing cert-based service principal auth.
 
+Log in using a Device:
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE=DEVICE to initiate Device login.
+   - azcopy login
+   Please note that you will be provided with a code to authenticate via a web browser. For example, you may see a message like this:
+   To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code AWRKVXE8P.
```
Comment (schoag-msft) [inline cmd/helpMessages.go:330]: Change the code to something generic - e.g. `ABCD12345`

File: cmd/helpMessages.go [hunk 4]
```diff
@@ -323,6 +323,21 @@ Set the environment variable AZCOPY_SPA_CERT_PASSWORD to the certificate's passw
...
+
+Log in using Managed Identity
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE to MSI.
+Set additional parameters based on your identity type:
+To use Client ID, set AZCOPY_MSI_CLIENT_ID.
+To use Object ID, set AZCOPY_MSI_OBJECT_ID.
+To use Resource string, set AZCOPY_MSI_RESOURCE_STRING.
+   - azcopy login
+   Upon successful authentication, you will see messages indicating login with identity succeeded and authenticating to the destination using Azure AD.
```
Comment (schoag-msft) [inline cmd/helpMessages.go:339]: Change `Azure AD` to `Microsoft Entra ID`

File: cmd/helpMessages.go (hunk 3 above)
Comment (pranavmalik-msft) [reply cmd/helpMessages.go]: Done in the latest commit.

File: cmd/helpMessages.go (hunk 1 above)
Comment (pranavmalik-msft) [reply cmd/helpMessages.go]: Done in the latest commit.

File: cmd/helpMessages.go [hunk 5]
```diff
@@ -313,6 +313,21 @@ Set the environment variable AZCOPY_SPA_CERT_PASSWORD to the certificate's passw
...
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE=DEVICE to initiate Device login.
+   - azcopy login
+   Please note that you will be provided with a code to authenticate via a web browser. For example, you may see a message like this:
+   To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code AWRKVXE8P.
+
+Log in using Managed Identity
+Set the environment variable AZCOPY_AUTO_LOGIN_TYPE to MSI.
+Set additional parameters based on your identity type:
+To use Client ID, set AZCOPY_MSI_CLIENT_ID.
+To use Object ID, set AZCOPY_MSI_OBJECT_ID.
```
[the remark this answers is not in the document]
Comment (pranavmalik-msft) [reply cmd/helpMessages.go]: Done in the latest commit.

File: cmd/helpMessages.go (hunk 4 above)
Comment (pranavmalik-msft) [reply cmd/helpMessages.go]: Done in the latest commit.

File: cmd/helpMessages.go [hunk 6]
```diff
@@ -323,6 +323,20 @@ Set the environment variable AZCOPY_SPA_CERT_PASSWORD to the certificate's passw
    Please treat /path/to/my/cert as a path to a PEM or PKCS12 file-- AzCopy does not reach into the system cert store to obtain your certificate.
    --certificate-path is mandatory when doing cert-based service principal auth.
 
+Log in using a Device:
```
Comment (vibhansa-msft) [inline cmd/helpMessages.go:326]: We shall get this part reviewed by Scott as well, as this is something visible to customer.

**Review by vibhansa-msft — APPROVED** [review]

File: cmd/helpMessages.go (hunk 6 above)
Comment (schoag-msft) [reply cmd/helpMessages.go:326]: lgtm

**Comment by schoag-msft** [comment]
lgtm

**Pull request merged**